### DISCUSSION TITLE
Transaction signing process

### DISCUSSION THREAD

**Discussion by pjmssb** [Cyfrin/foundry-full-course-cu#2989] [Q&A]
I have seen blockchain technology before, but I didn't learn to program in solidity, so I am looking at the Youtube version of **Learn Solidity, Blockchain Development, & Smart Contracts | Powered By AI - Full Course (0 - 6)**. In the 1:39:36 minute, talking about signing a transaction, it is described as "Someone with a private key signs a transaction by their private key being hashed with their transaction data". 

This may be not correct. Private/public keys are used with asymmetric encryption algorithms that are computationally expensive and have some limitations in the size of the content to encrypt. I believe that the signing may consist in encrypting the hash of the transactions with the private key. To verify the signing then will be to decrypt with the public key and you obtain the same hash, ergo it is verified. Isn't it? I am worng?

I didn't find this mentioned in other posts.

**Comment by EngrPips** [comment]
I think he meant the hash of the transaction data when he said `transaction data`
